src/binder/bind/bind_ddl.cpp

```diff
@@ -442,7 +423,16 @@ std::unique_ptr<BoundStatement> Binder::bindAddProperty(const Statement& stateme
     auto info = alter.getInfo();
     auto extraInfo = ku_dynamic_cast<ExtraAlterInfo*, ExtraAddPropertyInfo*>(info->extraInfo.get());
     auto tableName = info->tableName;
-    auto dataType = LogicalType::fromString(extraInfo->dataType);
+    auto dataType =
+        clientContext->getCatalog()->getType(clientContext->getTx(), extraInfo->dataType);
+    // TODO(Ziyi): Should we resolve the udt for default value there?
+    // or pass the clientContext to the transformer so we can solve at transforming time.
+    if (extraInfo->defaultValue->getExpressionType() == ExpressionType::LITERAL) {
```

Comment:
Currently we make default value an expression evaluator at runtime. Can we do the same for UDT? If not, better disable the default for UDT here.